homeassistant/components/sensor/seventeentrack.py

```diff
@@ -0,0 +1,236 @@
+CONF_SHOW_DELIVERED = 'show_delivered'
+
+DATA_PACKAGES = 'package_data'
+DATA_SUMMARY = 'summary_data'
+
+DEFAULT_ATTRIBUTION = 'Data provided by 17track.net'
+DEFAULT_SCAN_INTERVAL = timedelta(minutes=10)
+
+VALUE_DELIVERED = 'Delivered'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_SHOW_ARCHIVED, default=False): cv.boolean,
+    vol.Optional(CONF_SHOW_DELIVERED, default=False): cv.boolean,
+})
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Configure the platform and add the sensors."""
+    from py17track import Client
+    from py17track.errors import SeventeenTrackError
+
+    websession = aiohttp_client.async_get_clientsession(hass)
+
+    client = Client(websession)
+
+    try:
+        login_result = await client.profile.login(
+            config[CONF_USERNAME], config[CONF_PASSWORD])
+
+        if login_result is False:
+            _LOGGER.error('Invalid username and password provided')
+            return False
+    except SeventeenTrackError as err:
+        _LOGGER.error('There was an error while logging in: %s', err)
+        return False
+
+    if CONF_SCAN_INTERVAL in config:
+        scan_interval = config[CONF_SCAN_INTERVAL]
+    else:
+        scan_interval = DEFAULT_SCAN_INTERVAL
+
+    data = SeventeenTrackData(
+        client, scan_interval, config[CONF_SHOW_ARCHIVED],
+        config[CONF_SHOW_DELIVERED])
+    await data.async_update()
+
+    sensors = []
+
+    for status, quantity in data.summary.items():
+        sensors.append(SeventeenTrackSummarySensor(data, status, quantity))
+
+    for package in data.packages:
+        sensors.append(SeventeenTrackPackageSensor(data, package))
+
+    async_add_entities(sensors, True)
+
+
+class SeventeenTrackSummarySensor(Entity):
+    """Define a summary sensor."""
+
+    def __init__(self, data, name, initial_state):
+        """Initialize."""
+        self._attrs = {ATTR_ATTRIBUTION: DEFAULT_ATTRIBUTION}
+        self._data = data
+        self._name = name
+        self._state = initial_state
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        return self._attrs
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return 'mdi:package'
+
+    @property
+    def name(self):
+        """Return the name."""
+        return 'Packages {0}'.format(self._name)
+
+    @property
+    def state(self):
+        """Return the state."""
+        return self._state
+
+    @property
+    def unique_id(self):
+        """Return a unique, HASS-friendly identifier for this entity."""
+        return '{0}_{1}'.format(self._data.account_id, slugify(self._name))
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        if self._state == 1:
+            return 'package'
```

Comment:
Changing unit of measurement depending on sensor value is really unfriendly if using this with 3rd party data loggers.

Reply:
Got it. I'll forgo my English preferences in favor of data cleanliness. 😄